### DISCUSSION TITLE
Ignore .git file during chezmoi add -r dir_name/

### DISCUSSION THREAD

**Discussion by higoswami** [twpayne/chezmoi#3872] [Q&A] [answered]
I'm trying to add my neovim config using : `chezmoi add -r ~/.config/nvim` but  it adds everything from my nvim directory.  

I want to ignore `.git` and some other files from `~/.config/nvim` directory during `chezmoi add`. I tried `.chezmoiignore` but seems like it is used for `chezmoi apply` (not `chezmoi add`)

**Comment by Lockszmith-GH** [comment]
This is exactly what [`.chezmoiignore` is for](https://www.chezmoi.io/reference/special-files-and-directories/chezmoiignore/), it works with `add` as well. Note that it is slightly different than `.gitignore`.

If you share your `.chezmoiignore`, we can probably help you modify it to a functional one.

Also, check past [discussions](https://github.com/twpayne/chezmoi/discussions?discussions_q=ignore+in%3Atitle).

**Reply by higoswami** [reply]
I tried, but it didn't work. 

I made a new file (`.chezmoiignore`) at `~/.local/share/chezmoi`, then put the following content in it :

```
.git/ # ignore .git in any target directory
.gitignore
!~/.config/nvim/.gitignore

~/.config/nvim/lazyvim.json
~/.config/nvim/lazy-lock.json
```

Then i added the neovim config using : ` chezmoi add -r ~/.config/nvim` 
- Now when i went to ` ~/.local/share/chezmoi/dot_config/nvim` I can see that ignored files are there as well

Also `chezmoi ignored` command is not showing anything

Am i missing something ?

**Reply by bradenhilton** [reply, marked as answer]
@himanshugoswamiii You need something like

```text
**/.git/
**/.git/**
**/.gitignore
!.config/nvim/.gitignore

.config/nvim/lazyvim.json
.config/nvim/lazy-lock.json
```

**Reply by Lockszmith-GH** [reply]
Yep, as @bradenhilton demonstrated, you're version was incorrect.

`.chezmoiignore`'s root is you home, or more precisely, the home you'll preserve in the cache.
The `~/` in your initial version is what caused the issue.

**Reply by higoswami** [reply]
Thanks @bradenhilton  and @Lockszmith-GH . I got it all working now
